Source/Frontend/UI/UICore.cs

```diff
@@ -320,14 +299,6 @@ public static Form[] AllColorizedSingletons(Type baseType = null)
 
             isClosing = true;
 
-            foreach (Form frm in AllColorizedSingletons())
```

Comment:
This change is the most concerning to me. In the past we've relied on `IAutoColorize` to track open forms in order to close them.

Is this still necessary? If so, what method should be replace this with?

Reply:
This is a pretty bad hack that shouldn't be needed. Calling quit on the main window should message all forms that they should close. I don't think this is necessary and if anything is hanging because the message isn't making it up the chain, that should be fixed in individual instances. I'll remove it and test with it removed.

Reply:
Looks like Application.Exit() on the core form isn't completely closing all the forms as you'd expect it to (`Application.Exit()` signals to all forms that they should close, we shouldn't need that `Environment.Exit()` that's in there). I'll figure out what's hanging up so we can get rid of this.

Reply:
Resolved with https://github.com/redscientistlabs/RTCV/commit/a8b157a7fd773e93927451d0c581bfe73244a4e0

Reply:
I've also rebased the branch with this change removed for you.